File: Controllers/Api/TripsController.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using AutoMapper;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using TheWorld.Models;
using TheWorld.ViewModels;

namespace TheWorld.Controllers.Api
{
    [RouteAttribute("api/trips")]
    public class TripsController : Controller
    {
        private IWorldRepository _repository;
        private ILogger<TripsController> _logger;

        public TripsController(IWorldRepository repository, ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;
        }
        [HttpGetAttribute("")]
        // public JsonResult Get()   // Return Json can not return error 
        // {
        //     return Json(new Trip() { Name = "My Trip" });
        // }
        public IActionResult Get()
        {
            try
            {
                var result = _repository.GetAllTrips();
                return Ok(Mapper.Map<IEnumerable<TripVM>>(result));
            }
            catch (Exception exp)
            {
                _logger.LogError($"Failed to get all trips : {exp}");
                return BadRequest(exp);
            }
        }

        [HttpPostAttribute("")]
        public async Task<IActionResult> Post([FromBody]TripVM theTrip)
        {
            if (ModelState.IsValid)
            {
                // Save to database
                var newTrip = Mapper.Map<Trip>(theTrip);
                _repository.AddTrip(newTrip);
                if (await _repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripVM>(newTrip));
                }
            }
            return BadRequest("Failed to save changed to the database");
        }
    }
}

File: wwwroot/js/tripsController.js
// tripsController.js
(function () {
    "use strict";

    angular.module("app-trips").controller("tripsController", tripsController);

    function tripsController($http) {
        var vm = this;
        vm.trips = [];
        vm.newTrip = {};
        vm.errorMessage = "";
        vm.isBusy = true;

        $http.get("/api/trips")
            .then(function (response) {
                // success
                angular.copy(response.data, vm.trips);
                console.log(vm.trips);
            }, function () {
                // Failure
                vm.errorMessage = "Failed to load data: " + error;
            })
            .finally(function () {
                vm.isBusy = false;
            });

        // Create a new trip    
        vm.addTrip = function () {
            vm.isBusy = true;
            vm.errorMessage = "";

            $http.post("/api/trips", vm.newTrip)
                .then(function (response) {
                    // success
                    vm.trips.push(response.data);
                    vm.newTrip = {};
                    // Test
                    // vm.trips.push({name:vm.newTrip.name, created:new Date()});
                }, function () {
                    vm.errorMessage = "Failed to save new trip. " + error;
                })
                .finally(function () {
                    vm.isBusy = false;
                });


        }   
    }
})();